Undefined name: make_env in ./server/src/prefect_server/cli/dev.py

## Description
*A clear description of the bug*

An _undefined name_ like #2235 and #1199
https://github.com/PrefectHQ/prefect/blob/master/server/src/prefect_server/cli/dev.py#L88

`make_env` is an undefined name in this context which will raise a `NameError` at runtime.  Should this be `make_dev_env()` defined on line 36 or is `from prefect.cli.server import make_env` the right solution?

[flake8](http://flake8.pycqa.org) testing of https://github.com/PrefectHQ/prefect on Python 3.8.3

$ __flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics__
```
./server/src/prefect_server/cli/dev.py:88:11: F821 undefined name 'make_env'
    env = make_env()
          ^
1     F821 undefined name 'make_env'
1
```
https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead these tests are focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python a __NameError__ is raised which will halt/crash the script on the user.




## Expected Behavior
*What did you expect to happen instead?*




## Reproduction
*A minimal example that exhibits the behavior.*




## Environment
*Any additional information about your environment*

*Optionally run `prefect diagnostics` from the command line and paste the information here*

--- a/server/src/prefect_server/cli/dev.py
+++ b/server/src/prefect_server/cli/dev.py
@@ -85,7 +85,7 @@ def build(version):
     """
     docker_dir = Path(prefect_server.__file__).parents[2] / "docker"
 
-    env = make_env()
+    env = make_dev_env()
 
     if "PREFECT_SERVER_TAG" not in env:
         env.update(PREFECT_SERVER_TAG=version)

